src/Tvheadend.cpp

```diff
@@ -1385,13 +1385,9 @@ void CTvheadend::ParseRecordingAddOrUpdate ( htsmsg_t *msg, bool bAdd )
   {
     UPDATE(rec.state, PVR_TIMER_STATE_COMPLETED);
   }
-  else if (strstr(state, "missed") != NULL)
-  {
-    UPDATE(rec.state, PVR_TIMER_STATE_ERROR);
-  }
-  else if (strstr(state, "invalid")   != NULL)
+  else
   {
-    UPDATE(rec.state, PVR_TIMER_STATE_ERROR);
+    UPDATE(rec.error, state);
```

Comment:
Instead of removing the lines dealing with "missed" and "invalid" you should add

<pre>
else
{
  UPDATE(rec.state, PVR_TIMER_STATE_ERROR);
}  
</pre>

as a last resort for unhandled/unkown states (maybe introduced by new tvh versions).

Even more, this "else" is needed as you changed init value of SRecording::state to PVR_TIMER_STATE_NEW which would lead all recordings with unknown/unhandled as NEW. Unfortunately there is no PVR_TIMER_STATE_UNKNOWN. Thus we need to translate those unknown states to PVR_TIMER_STATE_ERROR.